http3/body.go

```diff
@@ -60,8 +71,30 @@ func (r *body) readImpl(b []byte) (int, error) {
 			}
 			switch f := frame.(type) {
 			case *headersFrame:
-				// skip HEADERS frames
-				continue
+				if r.isRequest {
+					// skip HEADERS frame in request
+					continue
+				}
+				if f.Length > r.maxHeaderBytes {
+					return 0, fmt.Errorf("HEADERS frame too large: %d bytes (max: %d)", f.Length, r.maxHeaderBytes)
+				}
+				p := make([]byte, f.Length)
+				r.str.Read(p)
+				trailers, err := r.decoder.DecodeFull(p)
+				if err != nil {
+					// Ignoring invalid frame
+					continue
+				}
+
+				for _, trailer := range trailers {
+					if r.resp.Trailer == nil {
+						r.resp.Trailer = http.Header{}
+					}
+					r.resp.Trailer.Add(trailer.Name, trailer.Value)
+				}
+
+				// Trailer Frame is the last frame.
+				return 0, nil
```

Comment:
I'm not sure if returning here is the right thing to do. Shouldn't we continue reading from the stream until we get an EOF, and throw an error if there's any other frame after the trailers?
Furthermore, would it be valid to send two HEADER frames that both contain trailers or is this not possible?

Reply:
https://tools.ietf.org/html/rfc7540#section-8.1
```
optionally, one HEADERS frame, followed by zero or more
       CONTINUATION frames containing the trailer-part, if present
```
So for me, if I understand well, it's only one HEADERS frame.

For the return, when I return it will make a `requestDone` and I thought it was enough, but I need to analyse more to be sure.

I took a look at the http2 implementation in the standard library, and, they control that there is an `EndStream` in the `HeaderFrame`, if there is not they raise an error, and then they `endStream` the current stream. I can do the same and it will be enough. WDYT?